src/library/compiler/llvm.cpp

```diff
@@ -0,0 +1,1639 @@
+    LLVMValueRef *args_carr = array_ref_to_ArrayLLVMValue(args_array_ref);
+    LLVMValueRef out =
+        LLVMConstArray(lean_to_Type(elemty), args_carr, args_array_ref.size());
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "...%s ; val: %s\n", __PRETTY_FUNCTION__,
+                LLVMPrintValueToString(out));
+    }
+    free(args_carr);
+    return lean_io_result_mk_ok(lean_box_usize(Value_to_lean(out)));
+#endif  // LEAN_LLVM
+}
+
+extern "C" LEAN_EXPORT lean_object *lean_llvm_const_string(
+    size_t ctx, lean_object *s, lean_object * /* w */) {
+    lean::string_ref sref = lean::string_ref(s, true);
+#ifndef LEAN_LLVM
+    lean_always_assert(
+        false && ("Please build a version of Lean4 with -DLLVM=ON to invoke "
+                  "the LLVM backend function."));
+#else
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "%s ; s: %s\n", __PRETTY_FUNCTION__, sref.data());
+    }
+
+    LLVMValueRef out =
+        LLVMConstStringInContext(lean_to_Context(ctx), sref.data(),
+                                 sref.length(), /*DontNullTerminate=*/false);
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "...%s ; val: %s\n", __PRETTY_FUNCTION__,
+                LLVMPrintValueToString(out));
+    }
+    return lean_io_result_mk_ok(lean_box_usize(Value_to_lean(out)));
+#endif  // LEAN_LLVM
+}
+
+extern "C" LEAN_EXPORT lean_object *lean_llvm_const_pointer_null(
+    size_t elemty, lean_object * /* w */) {
+#ifndef LEAN_LLVM
+    lean_always_assert(
+        false && ("Please build a version of Lean4 with -DLLVM=ON to invoke "
+                  "the LLVM backend function."));
+#else
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "%s ; elemty: %s\n", __PRETTY_FUNCTION__,
+                LLVMPrintTypeToString(lean_to_Type(elemty)));
+    }
+
+    LLVMValueRef out = LLVMConstPointerNull(lean_to_Type(elemty));
+
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "...%s ; out: %s\n", __PRETTY_FUNCTION__,
+                LLVMPrintValueToString(out));
+    }
+    return lean_io_result_mk_ok(lean_box_usize(Value_to_lean(out)));
+#endif  // LEAN_LLVM
+}
+
+extern "C" LEAN_EXPORT lean_object *llvm_get_param(size_t f, uint64_t ix,
+                                                   lean_object * /* w */) {
+#ifndef LEAN_LLVM
+    lean_always_assert(
+        false && ("Please build a version of Lean4 with -DLLVM=ON to invoke "
+                  "the LLVM backend function."));
+#else
+    if (LLVM_DEBUG) {
+        fprintf(stderr, "%s ; f: %s\n", __PRETTY_FUNCTION__,
+                LLVMPrintValueToString(lean_to_Value(f)));
+        fprintf(stderr, "...%s ; ix: %lu\n", __PRETTY_FUNCTION__, ix);
+    }
+    LLVMValueRef out = LLVMGetParam(lean_to_Value(f), ix);
+    fprintf(stderr, "%s ; out: %s\n", __PRETTY_FUNCTION__,
+            LLVMPrintValueToString(out));
+    return lean_io_result_mk_ok(lean_box_usize(Value_to_lean(out)));
+#endif  // LEAN_LLVM
+}
+
+extern "C" LEAN_EXPORT uint64_t llvm_count_params(size_t f,
```

Comment:
```suggestion
extern "C" LEAN_EXPORT uint64_t llvm_count_params(size_t /* ctx */, size_t f,
```
I'm pretty sure the implicit argument is passed as an argument to the C function.

You can check the C function signature the compiler expects by reading the `LLVMBindings.c` file.